### PR TITLE
Edit Product: short description

### LINKED ISSUE #1831: Edit Product: short description
Data:
- API field: `short_description`
- `Product` read-only model field: `briefDescription: String?`
---
- [ ] In the Edit Product main screen, add a row to display the current short description (or the placeholder, as shown in the screenshot below)
- [ ] Tapping the row above navigates to the Aztec editor for the short description since the content is in HTML

Screen Shot 2020-02-06 at 6 20 56 PM

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #1831 

## Changes

- In Networking layer, encoded `briefDescription` field
- Added a `UIImage` icon `briefDescriptionImage` + test case in `IconsTests`
- In `EditorFactory`, created an editor factory method for editing Product brief description
- In `DefaultProductFormTableViewModel` where we set up the sections and rows for Product Form UI, DI'ed `FeatureFlagService` for a flag that indicates whether Edit Products M2 flag is on
  - Added configuration for the brief description row
  - Showed the brief description row if Edit Products M2 flag is on
  - Used `trimmedBriefDescription` to trim HTML tags for brief description data for this row
- In `ProductFormViewController`, handled the edit brief description action by navigating to the Aztec editor
- Added unit tests

## Testing

- Go to the Products tab
- Tap on a simple Product
- Tap on the "Short description" row
- Tap "<" in the navigation bar --> should navigate back to the Edit Product screen
- Tap on the "Short description" row
- Enter some content that is different from before
- Tap "<" in the navigation bar --> should see an action sheet about whether to discard the changes
- Tap "Cancel" on the action sheet
- Tap "Done" in the navigation bar --> should navigate back to the Edit Product screen, and the short description updated
- Tap "Update"  in the navigation bar --> after the remote update is complete, check on the website to see if the short description is updated

## Example screenshots

no short description yet | edit short description | short description updated
-- | -- | --
Simulator Screen Shot - iPhone 11 - 2020-02-20 at 20 30 37 | Simulator Screen Shot - iPhone 11 - 2020-02-20 at 20 30 45 | Simulator Screen Shot - iPhone 11 - 2020-02-20 at 20 30 50

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by Garance91540** [comment]
👍

**Comment by jkmassel** [comment]
I'm freezing `3.7`, so this is being bumped to `3.8`. If you need it to be part of the `3.7` release, please re-target `release/3.7` and merge there. Once you have, please ping me that it's been merged and I'll be happy to cut a new beta release!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift [hunk 1]
```diff
@@ -47,18 +47,17 @@ private extension DefaultProductFormTableViewModel {
...
+        let rows: [ProductFormSection.SettingsRow?] = [
             .price(viewModel: priceSettingsRow(product: product)),
-            .shipping(viewModel: shippingSettingsRow(product: product)),
-            .inventory(viewModel: inventorySettingsRow(product: product))
+            shouldShowShippingSettingsRow ? .shipping(viewModel: shippingSettingsRow(product: product)): nil,
+            .inventory(viewModel: inventorySettingsRow(product: product)),
+            shouldShowBriefDescriptionRow ? .briefDescription(viewModel: briefDescriptionRow(product: product)): nil
         ]
+
+        return rows.compactMap { $0 }
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift]: Nice idea to use the compact map here to make the code shorted 👍

**Comment by jaclync** [comment]
> I've just a nit. I think that we can change the term brief description to short description everywhere. It can be a good improvement for legibility. In this way, we don't need to explain everywhere that when we are using briefDescription we are talking about the short description. But maybe there is a reason I don't know behind this choice. 😄 Feel free to choose

yea I thought about renaming `briefDescription` to `shortDescription`, but it's also in the Core Data schema and I'm not sure if a migration is worthwhile for this case (renaming an attribute requires a manual migration by creating a mapping model). alternatively, we could rename everything except for the Storage attribute - we have a little inconsistency in the naming, but maybe the readability would help more than the inconsistency? I created an issue for it https://github.com/woocommerce/woocommerce-ios/issues/1899, lemme know what you think and I can work on it since it's a quick task!

**Pull request merged**